### ISSUE TITLE
User list search - find by gpgkey

### ISSUE THREAD

**Issue by elhoim** [MISP/MISP#515]
The search filter builder for user list does not have gpgkey in the list of fields to create the request.

**Comment by iglocska** [comment]
I suppose searching for the keys would not be that interesting, but being able to search on whether a key is set or not would be, right?

**Comment by elhoim** [comment]
Yes only if the key is set or not, sorry for the lack of details.

**Comment by Rafiot** [comment]
Would also be nice to have a field to do a full text search from the list user page in order to find org names, or users. I'm not saying that people/orgs who already have access ask again because they forgot they have, but....

**Comment by iglocska** [comment]
Haha, sure. You can use the filters for now, but indeed, you'd have to
search for either org names or e-mail addresses, not both in one querry.
I'll add a similar free text search as what's on the event index.

**Comment by elhoim** [comment]
Confirmed still missing in current 2.5: the user index filter builder's field lists are `$booleanFields = array('autoalert', 'contactalert', 'termsaccepted', 'disabled', 'inactive')` and `$textFields = array('role', 'email')` at `app/Controller/UsersController.php:557-558` — `gpgkey` is in neither.

Opened MISP/MISP#10977, which implements exactly the narrowed ask from this thread (filter on whether a key is set, not a search over key material): `gpgkey` becomes a yes/no filter, mirroring the existing `inactive` precedent. Not runtime-verified — no live instance available on my side.

Side note on the other request in this thread: the free-text search over the user index does exist today — `searchall` / the `value` parameter matches e-mail, organisation name and role name (`app/Controller/UsersController.php:394` and `:441-449`).
